BrowserKit/Sources/SiteImageView/SiteImageHandler.swift

```diff
@@ -102,14 +109,25 @@ public class DefaultSiteImageHandler: SiteImageHandler {
 
     private func getFaviconImage(imageModel: SiteImageModel) async -> UIImage {
         do {
+            while let currentSiteRequest = currentInFlightRequest,
+               imageModel.siteURLString == currentSiteRequest {
+                // We are already processing a favicon request for this site
+                // Sleep this task until the previous request is completed
+                try? await Task.sleep(nanoseconds: 50_000_000) // 50ms
+            }
```

Comment:
@OrlaM @nbhasin2 This isn't particularly elegant but it should fix the redundant requests both for the site HTML and favicons and is the simplest fix I could find that also didn't cause any other obvious regressions. If anyone has suggestions on a better solution, however, please LMK. Happy to sync up or chat further.

Reply:
Is the source of the problem that one SiteImageView is trying to request the favicon multiple times or that multiple SiteImageViews are fetching the same favicon?
If the former then at what point is the current solution that is supposed to catch this failing? It's supposed to check against the currentURLString before it even gets to the image handler.
If it's the later then this fix won't help because SiteImageHandler is created new for each SiteImageView.

Reply:
> If the former then at what point is the current solution that is supposed to catch this failing? It's supposed to check against the currentURLString before it even gets to the image handler.

So this is a little bit more complicated than it seems, there is some code that is calling manuallySetImage repeatedly in the midst of also repeatedly updating the favicon to the same/identical URL. This results in the `currentURLString` getting reset basically so while it still serves a valid purpose, unfortunately the way that we're (over-)refreshing things still breaks it and causes redundant requests to be fired off.

> If it's the later then this fix won't help because SiteImageHandler is created new for each SiteImageView.

I've been testing this fairly exhaustively and if you put a breakpoint or log in the code which kicks off the network requests for the site HTML or the favicons, that code is called multiple times when first visiting a website, but with this change in place it is correctly only hit once (and subsequent requests wait until the original one is completed).